src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -151,6 +164,22 @@ protected override DbParameterCollection DbParameterCollection
         public override UpdateRowSource UpdatedRowSource { get; set; }
 
         /// <summary>
+        /// Releases any resources used by the connection and closes it.
+        /// </summary>
+        /// <param name="disposing">
+        /// true to release managed and unmanaged resources; false to release only unmanaged resources.
+        /// </param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                Connection?.DisposePreparedStatements(_preparedStatements);
+            }
```

Comment:
Instead of letting the connection own the statements, we may need to keep a weak reference to the commands and Dispose of any lingering ones up when the connection is closed.